contrib/core-contract-tests/tests/pox-4/pox_DelegateStackIncreaseCommand.ts

```diff
@@ -0,0 +1,132 @@
+import { expect } from "vitest";
+import { Cl } from "@stacks/transactions";
+
+/**
+ * The DelegateStackIncreaseCommand allows a pool operator to
+ * increase an active stacking lock, issuing a "partial commitment"
+ * for the increased cycles.
+ * 
+ * This method increases stacker's current lockup and partially
+ * commits the additional STX to pox-addr
+ *
+ * Constraints for running this command include:
+ * - The Stacker must have locked uSTX.
+ * - The Operator has to currently be delegated by the Stacker.
+ * - The increase amount must be greater than 0.
+ * - Stacker's unlocked uSTX amount must be greater than or equal
+ *   to the value of the increase amount.
+ * - Stacker's maximum delegated amount must be greater than or equal
+ *   to the final locked amount.
+ * - The Operator must have locked the Stacker's previously locked funds.
+ */
+export class DelegateStackIncreaseCommand implements PoxCommand {
+  readonly operator: Wallet;
+  readonly stacker: Wallet;
+  readonly increaseBy: bigint;
+
+  /**
+   * Constructs a DelegateStackIncreaseCommand to increase the uSTX amount
+   * previously locked on behalf of a Stacker.
+   *
+   * @param operator - Represents the Pool Operator's wallet.
+   * @param stacker - Represents the Stacker's wallet.
+   * @param increaseBy - Represents the locked amount to be increased by
+   */
+  constructor(operator: Wallet, stacker: Wallet, increaseBy: bigint) {
+    this.operator = operator;
+    this.stacker = stacker;
+    this.increaseBy = increaseBy;
+  }
+
+  check(model: Readonly<Stub>): boolean {
+    // Constraints for running this command include:
+    // - The Stacker must have locked uSTX.
+    // - The Operator has to currently be delegated by the Stacker.
+    // - The increase amount must be greater than 0.
+    // - Stacker's unlocked uSTX amount must be greater than or equal
+    //   to the value of the increase amount.
+    // - Stacker's maximum delegated amount must be greater than or equal
+    //   to the final locked amount.
+    // - The Operator must have locked the Stacker's previously locked funds.
+
+    const operatorWallet = model.wallets.get(this.operator.stxAddress)!;
+    const stackerWallet = model.wallets.get(this.stacker.stxAddress)!;
+
+    return (
+      stackerWallet.amountLocked > 0 &&
+      stackerWallet.hasDelegated === true &&
+      stackerWallet.isStacking === true &&
+      this.increaseBy > 0 &&
+      operatorWallet.poolMembers.includes(stackerWallet.stxAddress) &&
+      stackerWallet.amountUnlocked >= this.increaseBy &&
+      stackerWallet.delegatedMaxAmount >=
+        Number(this.increaseBy) + stackerWallet.amountLocked &&
+      operatorWallet.hasLocked.indexOf(stackerWallet.stxAddress) > -1
+    );
+  }
+
+  run(model: Stub, real: Real): void {
+    model.trackCommandRun(this.constructor.name);
+
+    const prevLocked = this.stacker.amountLocked;
+    const newTotalLocked = prevLocked + Number(this.increaseBy);
```

Comment:
Given the concerns with precision and overflow when converting BigInts to Numbers, it might be more effective to declare `increaseBy` as a Number from the start, provided its values reliably stay within the Number's safe integer range.